### PR TITLE
fix(ubuntu): removed version from kernel flavor

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind documentation

> /kind feature

**Any specific area of the project related to this PR?**

/area crawler

**What this PR does / why we need it**:

This removes a **redundant** information in the output kernel `flavor`: instead of maintaining different "subflavors" (e.g. `aws-5.15`, `aws-5.19`, ...), we only keep one flavor (e.g. `aws`). At the end we'll end up with:

```
aws-5.19 -> aws
aws-5.15 -> aws
azure-5.15 -> azure
azure-5.19 -> azure
...
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: kernel_crawler/ubuntu.py [hunk 1]
```diff
@@ -28,9 +28,11 @@ def to_driverkit_config(self, release, deps):
                 # if capture was successful, set the flavor
                 if flavor_capture is not None:
                     flavor = flavor_capture.group(1)  # set flavor to the first capture group
+                    if '-' in flavor:
+                        flavor = flavor.split('-')[0]
 
                 target = f'ubuntu-{flavor}'    # expose the correct ubuntu flavor
-                release = f'{krel}-{flavor}'  # add flavor to release
+                release = f'{krel}'
```
Comment (FedeDP) [inline kernel_crawler/ubuntu.py]: The `release` should keep the `-{flavor}` imho.

**Comment by FedeDP** [comment]
The issues this patch is trying to fix are:
* test-infra dbg is not building `ubuntu-aws-5.19_...` because it does not match its regex
* moreover, that `-5.19` suffix is not present elsewhere (`uname -r` or `uname -v` outputs)
* finally, we will also need a falco-driver-loader patch because we want to distinguish:
  * `ubuntu-aws_5.19.0-1025-aws_26~22.04.1`
  * `ubuntu-aws_5.19.0-1025-aws_26`
The latter is what we build today; the former is why we need a driverloader patch: basically, we will need to fixup the kernel-version to be `26~22.04.1` for these kind of drivers.

**Comment by FedeDP** [comment]
So, for the kenrel-crawler side, it is enough to drop the `-5.19` like suffixes from the ubuntu flavor.

**Comment by FedeDP** [comment]
See the falco-driver-loader patch: https://github.com/falcosecurity/falco/pull/2635

**Comment by therealbobo** [comment]
Moreover I'd like to add that the main problem was encountered with the ubuntu aws flavor on arm: it seems that the amd64 ubuntu-aws kernel is the same for the kernel version `#26` and `#26~22.04.1` but not for the arm64 kernels. Given that, as @FedeDP said, test-infra dbg is not building these kernel versions (e.g. `#26~22.04.1`).

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
@therealbobo can you rebase on top of main?

**Comment by FedeDP** [comment]
ah no need! tide is ok now

**Pull request merged**